File: app/android/app/src/main/java/com/magicvector/domain/convertor/UserConvertor.kt
package com.magicvector.domain.convertor

import com.magicvector.domain.entity.UserEntity
import com.magicvector.domain.model.UserSessionModel

object UserConvertor {
    fun model2Entity(session: UserSessionModel): UserEntity {
        return UserEntity(
            userId = session.userId,
            account = session.account,
            name = session.name,
            avatarUrl = session.avatarUrl,
            accessToken = session.accessToken,
            password = session.password,
            isCurrent = session.isCurrent,
            lastLoginAt = session.lastLoginAt
        )
    }

    fun entity2Model(entity: UserEntity): UserSessionModel {
        return UserSessionModel(
            userId = entity.userId,
            account = entity.account,
            name = entity.name,
            avatarUrl = entity.avatarUrl,
            accessToken = entity.accessToken,
            password = entity.password,
            isCurrent = entity.isCurrent,
            lastLoginAt = entity.lastLoginAt
        )
    }
}

File: demo/flutter/flutternew/lib/domain/convertor/user_convertor.dart
import '../entity/user_entity.dart';
import '../model/user_session_model.dart';

/// 对齐 Android [UserConvertor]。
class UserConvertor {
  UserConvertor._();

  static UserEntity modelToEntity(UserSessionModel m) {
    return UserEntity(
      userId: m.userId,
      account: m.account,
      name: m.name,
      avatarUrl: m.avatarUrl,
      accessToken: m.accessToken,
      password: m.password,
      isCurrent: m.isCurrent,
      lastLoginAt: m.lastLoginAt,
    );
  }

  static UserSessionModel entityToModel(UserEntity e) {
    return UserSessionModel(
      userId: e.userId,
      account: e.account,
      name: e.name,
      avatarUrl: e.avatarUrl,
      accessToken: e.accessToken,
      password: e.password,
      isCurrent: e.isCurrent,
      lastLoginAt: e.lastLoginAt,
    );
  }

  static UserEntity fromRow(Map<String, Object?> row) => UserEntity.fromRow(row);

  static Map<String, Object?> entityToRow(UserEntity e) => e.toRow();
}
